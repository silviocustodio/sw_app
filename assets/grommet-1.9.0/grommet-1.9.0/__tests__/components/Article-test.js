// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Article from '../../src/js/components/Article';
import Heading from '../../src/js/components/Heading';
import Section from '../../src/js/components/Section';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Article', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Article>
        <Heading>
          Title
        </Heading>
        <Section>
          <h2>
            Heading
          </h2>
          <p>
            Lorem ipsum ...
          </p>
        </Section>
      </Article>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a classname', () => {
    const component = renderer.create(
      <Article className="test">
        <Heading>
          Title
        </Heading>
        <Section>
          <h2>
            Heading
          </h2>
          <p>
            Lorem ipsum ...
          </p>
        </Section>
      </Article>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders microdata properties', () => {
    const component = renderer.create(
      <Article itemScope={true} itemType="http://schema.org/Article" 
        itemProp="test">
        <Heading>
          Title
        </Heading>
        <Section>
          <h2>
            Heading
          </h2>
          <p>
            Lorem ipsum ...
          </p>
        </Section>
      </Article>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
