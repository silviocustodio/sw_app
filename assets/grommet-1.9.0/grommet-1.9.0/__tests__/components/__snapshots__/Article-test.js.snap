// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article has correct default options 1`] = `
<article
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-article"
  id={undefined}
  onClick={undefined}
  onScroll={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <h1
    className="grommetux-heading"
  >
    Title
  </h1>
  <section
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-vertical-medium grommetux-section"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <h2>
      Heading
    </h2>
    <p>
      Lorem ipsum ...
    </p>
  </section>
  <a
    aria-hidden="true"
    tabIndex="-1"
  />
</article>
`;

exports[`Article renders a classname 1`] = `
<article
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-article test"
  id={undefined}
  onClick={undefined}
  onScroll={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <h1
    className="grommetux-heading"
  >
    Title
  </h1>
  <section
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-vertical-medium grommetux-section"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <h2>
      Heading
    </h2>
    <p>
      Lorem ipsum ...
    </p>
  </section>
  <a
    aria-hidden="true"
    tabIndex="-1"
  />
</article>
`;

exports[`Article renders microdata properties 1`] = `
<article
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-article"
  id={undefined}
  itemProp="test"
  itemScope={true}
  itemType="http://schema.org/Article"
  onClick={undefined}
  onScroll={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <h1
    className="grommetux-heading"
  >
    Title
  </h1>
  <section
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-vertical-medium grommetux-section"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <h2>
      Heading
    </h2>
    <p>
      Lorem ipsum ...
    </p>
  </section>
  <a
    aria-hidden="true"
    tabIndex="-1"
  />
</article>
`;
